Makefile

```diff
@@ -976,8 +976,16 @@ export CC_FLAGS_LTO
 endif
 
 ifdef CONFIG_CFI_CLANG
-CC_FLAGS_CFI	:= -fsanitize=kcfi
-KBUILD_CFLAGS	+= $(CC_FLAGS_CFI)
+CC_FLAGS_CFI	 := -fsanitize=kcfi
+ifdef CONFIG_RUST
+# If Rust is enabled, this flag is required to support x-lang integers
+# This addresses the problem that on e.g. i686, int != long, and Rust
+# calls both i32.
```

Comment:
Perhaps:

```suggestion
# If Rust is enabled, this flag is required to support cross-language integer types.
# This addresses the problem that on e.g. i686, int != long, and Rust
# maps both to i32.
```

Or perhaps we could give a bit more details (or a link to those details), e.g. something like "... int and long are different types for CFI checking purposes, but Rust maps both to i32", assuming that makes sense (@rcvalle).

Reply:
@maurer as a suggestion, you may point to the Clang [-fsanitize-cfi-icall-experimental-normalize-integers](https://clang.llvm.org/docs/ControlFlowIntegrity.html#fsanitize-cfi-icall-experimental-normalize-integers) option description and the _Encoding C integer types_ section in the [design document](https://rcvalle.com/docs/rust-cfi-design-doc.pdf).